exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/Statement.kt

```diff
@@ -44,7 +44,9 @@ abstract class Statement<out T>(val type: StatementType, val targets: List<Table
         }
 
         val statement = try {
-            prepared(transaction, prepareSQL(transaction))
+            prepared(transaction, prepareSQL(transaction)).apply {
+                timeout = transaction.timeout
+            }
```

Comment:
I thought about injecting a timeout into the preparedStatement creation, but I wasn't sure if that would work, so this is how I implemented it. If there's a better way, I'll try it.

Reply:
Thanks for putting this together so quickly.

I agree that this way works best. Like this, we can choose later on to include a global default timeout setting in `DatabaseConfig` and set it here. Or we can extend the feature on a per-`Query` basis and override this setting in the class `executeInternal()`.

I have some comments about the tests and the transaction property, but I'll wait til you're ready for a review.

Btw, if it's possible and you're planning on adding tests to the spring or spring boot modules, please consider splitting this PR into 2 separate PRs. One for just the core and jdbc changes alone, then a second one after for the spring changes only.

Reply:
I'm happy to see your response quickly

I can put default values in the DatabaseConfig if needed. Let me know if you need to do that.

The test code for SpringTransactionManager will be to see if the timeout value is set correctly, but I'll split the PR and write some more.

Let me know what you think about the transaction property and I'll take a look.